### PR TITLE
PB-6302: Update sched-ops to reinitialise

### PR DESCRIPTION
**Pull request by kshithijiyer-px**
**What this PR does / why we need it**:
Testcases where there were more than one app deplymennt were failing with the below trace which was because we are trying to reuse the same client which is created for the earlier namespace:
```
[tests.processError:#635] - processError. Err: timed out performing task., Error was: DoRetryWithTimeout timed out. Errors generated in retries: {Failed to schedule app: tektoncd due to err: Failed to create task: git-clone, Err: the namespace of the provided object does not match the namespace sent on the request}
{Failed to schedule app: tektoncd due to err: Failed to create task: git-clone, Err: the namespace of the provided object does not match the namespace sent on the request}
{Failed to schedule app: tektoncd due to err: Failed to create task: git-clone, Err: the namespace of the provided object does not match the namespace sent on the request}
{Failed to schedule app: tektoncd due to err: Failed to create task: git-clone, Err: the namespace of the provided object does not match the namespace sent on the request}
{Failed to schedule app: tektoncd due to err: Failed to create task: git-clone, Err: the namespace of the provided object does not match the namespace sent on the request}
{Failed to schedule app: tektoncd due to err: Failed to create task: git-clone, Err: the namespace of the provided object does not match the namespace sent on the request}
{Failed to schedule app: tektoncd due to err: Failed to create task: git-clone, Err: the namespace of the provided object does not match the namespace sent on the request}
{Failed to schedule app: tektoncd due to err: Failed to create task: git-clone, Err: the namespace of the provided object does not match the namespace sent on the request}
{Failed to schedule app: tektoncd due to err: Failed to create task: git-clone, Err: the namespace of the provided object does not match the namespace sent on the request}
{Failed to schedule app: tektoncd due to err: Failed to create task: git-clone, Err: the namespace of the provided object does not match the namespace sent on the request}
{Failed to schedule app: tektoncd due to err: Failed to create task: git-clone, Err: the namespace of the provided object does not match the namespace sent on the request}
{Failed to schedule app: tektoncd due to err: Failed to create task: git-clone, Err: the namespace of the provided object does not match the namespace sent on the request}  
```
We need to do the below to fix the same:
- [x] Vendor in https://github.com/portworx/sched-ops/pull/363 to reinitialise client for each namespace
- [x] Skip Volume validation  in case of tektoncd app

Dashboard URL:
https://aetos.pwx.purestorage.com/resultSet/testSetID/589435 (Single testcase run)

**Which issue(s) this PR fixes** (optional)
Closes #PB-6302

**Special notes for your reviewer**:

### CODE REVIEW

File: tests/common.go [hunk 1]
```diff
@@ -699,6 +699,10 @@ func ValidateContext(ctx *scheduler.Context, errChan ...*chan error) {
 		}
 
 		Step(fmt.Sprintf("validate %s app's volumes", ctx.App.Key), func() {
+			// In case of tektoncd skip the volume validation
```
Comment (mkoppal-px) [inline tests/common.go:702]: Add the reason for skipping volume validation in the doc string

**Comment by mkoppal-px** [comment]
Can you run one test where we are taking backup from source and restoring it to destination? We need to check even for restored app contexts the ValidateContext is skipping volume validation.

File: tests/common.go (hunk 1 above)
Comment (kshithijiyer-px) [reply tests/common.go]: Done

**Comment by kshithijiyer-px** [comment]
> Can you run one test where we are taking backup from source and restoring it to destination? We need to check even for restored app contexts the ValidateContext is skipping volume validation.

i am planning to rerun the pipeline once the sched-ops change is merged. Will that do?

**Comment by kshithijiyer-px** [comment]
> Can you run one test where we are taking backup from source and restoring it to destination? We need to check even for restored app contexts the ValidateContext is skipping volume validation.

https://aetos.pwx.purestorage.com/resultSet/testSetID/589653  Have ran it with BasicBackupCreation and it's working fine.

**Comment by kshithijiyer-px** [comment]
Runs with master torpedo vendored in:
https://aetos.pwx.purestorage.com/resultSet/testSetID/590715 
https://aetos.pwx.purestorage.com/resultSet/testSetID/590838

File: tests/common.go [hunk 2]
```diff
@@ -699,6 +699,10 @@ func ValidateContext(ctx *scheduler.Context, errChan ...*chan error) {
 		}
 
 		Step(fmt.Sprintf("validate %s app's volumes", ctx.App.Key), func() {
+			// In case of tektoncd skip the volume validation as the pods are created through jobs and not deployments or sts
+			if Contains(Inst().AppList, "tektoncd") {
```
Comment (kphalgun-px) [inline tests/common.go:703]: How about?

```go
if strings.Contains(ctx.App.Key, "tektoncd") {
    ctx.SkipVolumeValidation = true
}
```

Comment (kshithijiyer-px) [reply tests/common.go]: Done

**Review by kphalgun-px — APPROVED** [review]
LGTM

**Review by mkoppal-px — APPROVED** [review]
LGTM

**Comment by kshithijiyer-px** [comment]
Dashboard URLs with the final branch:
-  https://aetos.pwx.purestorage.com/resultSet/testSetID/591958 
-  https://aetos.pwx.purestorage.com/resultSet/testSetID/591879

**Pull request merged**